e2etest/bgptests/bgp.go

```diff
@@ -662,6 +662,33 @@ var _ = ginkgo.Describe("BGP", func() {
 				return nil
 			}, 4*time.Minute, 1*time.Second).ShouldNot(HaveOccurred())
 
+			ginkgo.By("checking the sessions don't flap when changing the configuration")
+
+			previousNeighbors := map[string]frr.NeighborsMap{}
+			for _, c := range FRRContainers {
+				neighbors, err := frr.NeighborsInfoMap(c)
+				Expect(err).NotTo(HaveOccurred())
+				previousNeighbors[c.Name] = neighbors
+			}
+			ginkgo.By("creating another the service")
+			svc1, _ := testservice.CreateWithBackend(cs, testNamespace, "external-local-lb1", tweak)
+			testservice.Delete(cs, svc1)
+
+			Consistently(func() error {
+				for _, c := range FRRContainers {
+					neighbors, err := frr.NeighborsInfo(c)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(neighbors).To(HaveLen(len(previousNeighbors[c.Name])))
```

Comment:
key in map if not exists (not I can see that happen) panics

Reply:
yep but in this case I think it's fair because:
- it's a test, we'll see where it fails and eventually adjust it
- we are filling the map with the same set of keys we are using here, just a few lines above, so it shouldn't happen

All in all, the extra check is not worth the break in following the code imo

Reply:
in 10 years from now it could be many lines diff :)
we can let it or you can use the https://onsi.github.io/gomega/#havekeykey-interface is single line

Reply:
> in 10 years from now it could be many lines diff :) we can let it or you can use the https://onsi.github.io/gomega/#havekeykey-interface is single line

Is there any added value if we add the assertion vs having it explode? The test will fail anyway (even more elegantly) and it will still tell us the missing key.